nixos/modules/misc/nixpkgs.nix

```diff
@@ -324,74 +324,135 @@ in
-        # don't need to evaluate `finalPkgs`.
-        lib.mkOverride lib.modules.defaultOverridePriority
-          finalPkgs.__splicedPackages;
-    };
+        default = null;
 
-    assertions = let
-      # Whether `pkgs` was constructed by this module. This is false when any of
-      # nixpkgs.pkgs or _module.args.pkgs is set.
-      constructedByMe =
-        # We set it with default priority and it can not be merged, so if the
-        # pkgs module argument has that priority, it's from us.
-        (lib.modules.mergeAttrDefinitionsWithPrio options._module.args).pkgs.highestPrio
-          == lib.modules.defaultOverridePriority
-        # Although, if nixpkgs.pkgs is set, we did forward it, but we did not construct it.
-          && !opt.pkgs.isDefined;
-    in [
-      (
-        let
-          nixosExpectedSystem =
-            if config.nixpkgs.crossSystem != null
-            then config.nixpkgs.crossSystem.system or (lib.systems.parse.doubleFromSystem (lib.systems.parse.mkSystemFromString config.nixpkgs.crossSystem.config))
-            else config.nixpkgs.localSystem.system or (lib.systems.parse.doubleFromSystem (lib.systems.parse.mkSystemFromString config.nixpkgs.localSystem.config));
-          nixosOption =
-            if config.nixpkgs.crossSystem != null
-            then "nixpkgs.crossSystem"
-            else "nixpkgs.localSystem";
-          pkgsSystem = finalPkgs.stdenv.targetPlatform.system;
-        in {
-          assertion = constructedByMe -> !hasPlatform -> nixosExpectedSystem == pkgsSystem;
-          message = "The NixOS nixpkgs.pkgs option was set to a Nixpkgs invocation that compiles to target system ${pkgsSystem} but NixOS was configured for system ${nixosExpectedSystem} via NixOS option ${nixosOption}. The NixOS system settings must match the Nixpkgs target system.";
-        }
-      )
-      {
-        assertion = constructedByMe -> hasPlatform -> legacyOptionsDefined == [];
-        message = ''
-          Your system configures nixpkgs with the platform parameter${optionalString hasBuildPlatform "s"}:
-          ${hostPlatformLine
-          }${buildPlatformLine
-          }
-          However, it also defines the legacy options:
-          ${concatMapStrings showOptionWithDefLocs legacyOptionsDefined}
-          For a future proof system configuration, we recommend to remove
-          the legacy definitions.
+        description = mdDoc ''
+          The derivation to the nixpkgs sources, as a string with context. This
+          is set up by the nixpkgs flake.
+        '';
+      };
+
+      setNixPath = mkOption {
+        type = types.bool;
+
+        default = cfg.flake.source != null;
+        defaultText = "config.nixpkgs.flake.source != null";
+
+        description = mdDoc ''
+          Whether to create `/etc/nix/inputs/nixpkgs` as a symlink to the version of nixpkgs that
+          the NixOS system was built with and set `NIX_PATH` to include `/etc/nix/inputs/nixpkgs`.
```

Comment:
just to be absolutely clear: the symlink points to a path in the store, right?

Reply:
Yep. I will word this more carefully.